virttest/utils_net.py

```diff
@@ -1191,6 +1195,68 @@ def wrap_init(*args, **kargs):
     return wrap_init
 
 
+def setup_ovs_vhostuser(hp_num, tmpdir, br_name, port_names,
+                        queue_size=None):
+    """
+    Setup vhostuser interface with openvswitch and dpdk
+
+    :param hp_num: hugepage count number
+    :param tmpdir: tmp directory for save openvswitch test files
+    :param br_name: name of bridge
+    :param port_names: list name of port need to add to bridge
+    """
+    clean_ovs_env(selinux_mode="permissive", page_size=hp_num,
+                  clean_ovs=True)
+
+    # Install openvswitch
+    utils_package.package_install("openvswitch")
+
+    # Init ovs
+    ovs = factory(openvswitch.OpenVSwitch)(tmpdir)
+    ovs.init_new()
+    if not ovs.check():
+        raise Exception("Check of OpenVSwitch failed.")
+
+    # Create bridge and ports
+    ovs.create_br_port(br_name, port_names)
+
+    # Enable multiqueue size
+    if queue_size:
+        ovs.enable_multiqueue(port_names, int(queue_size))
+
+    return ovs
+
+
+def clean_ovs_env(run_dir="/var/run/openvswitch", selinux_mode=None,
+                  page_size=None, clean_ovs=False):
+    """
+    Cleanup ovs environment
+
+    :param run_dir: openvswitch run dir
+    :param selinux_mode: permissive or enforcing
+    :param page_size: size for setting hugepage
+    :param clean_ovs: clean ovs service or not
+    """
+    # Clean dir
+    if os.path.exists(run_dir):
+        shutil.rmtree(run_dir)
+    os.mkdir(run_dir)
```

Comment:
is any reason to create the folder again since it is cleanup logic?

Reply:
As the dir exist before the operation

Reply:
If you are sure the dir exists before the operation, do we need to check its existence in L1244?

Reply:
It just to make sure that the operation are done valuable